### PR TITLE
Fix windows builds

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
supersedes https://github.com/conda-forge/pytorch-cpu-feedstock/pull/63

**Comment by mikemhenry** [comment]
Okay got our first problem to fix:
```
Encountered problems while solving:
  - package liblief-0.12.0.dev20210731-h330b295_0 is excluded by strict repo priority
  - package py-lief-0.12.0.dev20210731-py37h7c545b2_0 is excluded by strict repo priority
```

**Comment by mikemhenry** [comment]
@hmaarrfk :) I was just about to ask you what was going on with that in the conda-forge.yml

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
I think we need this: https://github.com/conda-forge/lief-feedstock/pull/32 its finally time to move on from "dev"

**Comment by hmaarrfk** [comment]
Anyway, I also think it is a good idea to skip unix builds (and the repeated python builds) until you get to a working state.  Iguess I got the skip python syntax wrong.

**Comment by hmaarrfk** [comment]
Good luck

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

ok i'll stop messing around

**Comment by hmaarrfk** [comment]
i really dislike split package recipes.

**Comment by mikemhenry** [comment]
@hmaarrfk Ah yes that makes sense why it wasn't quite working... I couldn't figure it out either. 
Thanks for all your help! I will keep working on it and just ping you if I get stuck.

**Comment by mikemhenry** [comment]
looks like I need to rebase the patch

**Comment by mikemhenry** [comment]
Okay assuming my patch didn't break things, new error 
```
(D:\bld\pytorch-recipe_1659387233087\_build_env) (base) D:\bld\pytorch-recipe_1659387233087\work>D:\bld\pytorch-recipe_1659387233087\_h_env\python.exe setup.py install 
Building wheel torch-1.12.0.post2
CMake Error at CMakeLists.txt:23 (project):
-- Configuring incomplete, errors occurred!
  Generator
See also "D:/bld/pytorch-recipe_1659387233087/work/build/CMakeFiles/CMakeOutput.log".

-- Building version 1.12.0.post2
cmake -GNinja -DBUILD_CUSTOM_PROTOBUF=ON -DBUILD_PLATFORM=win-64 -DBUILD_PREFIX=D:\bld\pytorch-recipe_1659387233087\_build_env -DBUILD_PYTHON=True -DBUILD_TEST=False -DCMAKE_BUILD_TYPE=Release -DCMAKE_GEN=Visual Studio 16 2019 -DCMAKE_GENERATOR=Ninja -DCMAKE_GENERATOR_TOOLSET=v141 -DCMAKE_INCLUDE_PATH=D:\bld\pytorch-recipe_1659387233087\_h_env\Library\include -DCMAKE_INSTALL_PREFIX=D:\bld\pytorch-recipe_1659387233087\work\torch -DCMAKE_LIBRARY_PATH=D:\bld\pytorch-recipe_1659387233087\_h_env\Library\lib -DCMAKE_PLAT=x64 -DCMAKE_PREFIX_PATH=D:\bld\pytorch-recipe_1659387233087\_h_env\Lib\site-packages;D:\bld\pytorch-recipe_1659387233087\_h_env\Library -DINSTALL_TEST=0 -DNUMPY_INCLUDE_DIR=D:\bld\pytorch-recipe_1659387233087\_h_env\lib\site-packages\numpy\core\include -DPYTHON_EXECUTABLE=D:\bld\pytorch-recipe_1659387233087\_h_env\python.exe -DPYTHON_INCLUDE_DIR=D:\bld\pytorch-recipe_1659387233087\_h_env\Include -DPYTHON_LIBRARY=D:\bld\pytorch-recipe_1659387233087\_h_env/libs/python38.lib -DTORCH_BUILD_VERSION=1.12.0.post2 -DUSE_CUDA=0 -DUSE_MKLDNN=0 -DUSE_NEW_CMAKE_GEN_SYNTAX=1 -DUSE_NUMPY=True -DUSE_OPENMP=0 -DUSE_SYSTEM_SLEEF=ON D:\bld\pytorch-recipe_1659387233087\work
    Ninja

  does not support toolset specification, but toolset

    v141

  was specified.


CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
```

**Comment by mikemhenry** [comment]
pytorch needs vs2019 https://github.com/pytorch/pytorch#from-source
lets see if this works!

**Comment by mikemhenry** [comment]
well it "works"
```
Generator

    Ninja

  does not support toolset specification, but toolset

    v142

  was specified.
```

Now it says v142 was specified instead of v141, but it still doesn't know how to handle that

**Comment by mikemhenry** [comment]
```
FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/remove_expands.cpp.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DAT_PER_OPERATOR_HEADERS -DCPUINFO_SUPPORTED_PLATFORM=1 -DFMT_HEADER_ONLY=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DUSE_C10D_GLOO -DUSE_DISTRIBUTED -DUSE_EXTERNAL_MZCRC -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE=1 -D_OPENMP_NOFORCE_MANIFEST -Dtorch_cpu_EXPORTS -ID:\bld\pytorch-recipe_1663314310088\work\build\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\build -ID:\bld\pytorch-recipe_1663314310088\work -ID:\bld\pytorch-recipe_1663314310088\work\third_party\onnx -ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\onnx -ID:\bld\pytorch-recipe_1663314310088\work\third_party\foxi -ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\foxi -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\api -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\api\include -ID:\bld\pytorch-recipe_1663314310088\work\caffe2\aten\src\TH -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\aten\src\TH -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\..\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc -ID:\bld\pytorch-recipe_1663314310088\work\third_party\miniz-2.0.8 -ID:\bld\pytorch-recipe_1663314310088\work\third_party\kineto\libkineto\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\kineto\libkineto\src -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\distributed -ID:\bld\pytorch-recipe_1663314310088\work\aten\..\third_party\catch\single_include -ID:\bld\pytorch-recipe_1663314310088\work\aten\src\ATen\.. -ID:\bld\pytorch-recipe_1663314310088\work\c10\.. -ID:\bld\pytorch-recipe_1663314310088\work\third_party\pthreadpool\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\cpuinfo\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm\third_party\asmjit\src -ID:\bld\pytorch-recipe_1663314310088\work\third_party\FP16\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fmt\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\flatbuffers\include -external:ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\gloo -external:ID:\bld\pytorch-recipe_1663314310088\work\cmake\..\third_party\gloo -external:ID:\bld\pytorch-recipe_1663314310088\work\third_party\protobuf\src -external:ID:\bld\pytorch-recipe_1663314310088\_h_env\Library\include -external:ID:\bld\pytorch-recipe_1663314310088\work\third_party\XNNPACK\include -external:ID:\bld\pytorch-recipe_1663314310088\work\cmake\..\third_party\eigen -external:ID:\bld\pytorch-recipe_1663314310088\work\caffe2 -external:W0 /DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2 /DNDEBUG /w /bigobj -DNDEBUG -DCAFFE2_USE_GLOO -DTH_HAVE_THREAD /EHsc /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -O2 -DCAFFE2_BUILD_MAIN_LIB -DONNX_BUILD_MAIN_LIB -std:c++14 /showIncludes /Focaffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_expands.cpp.obj /Fdcaffe2\CMakeFiles\torch_cpu.dir\ /FS -c D:\bld\pytorch-recipe_1663314310088\work\torch\csrc\jit\passes\remove_expands.cpp
D:\bld\pytorch-recipe_1663314310088\work\torch\csrc\jit\passes\remove_expands.cpp : fatal error C1085: Cannot write compiler generated file: 'D:\bld\pytorch-recipe_1663314310088\work\build\caffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_expands.cpp.obj': No space left on device
[4583/5100] Building CXX object caffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_mutation.cpp.obj
FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/remove_mutation.cpp.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DAT_PER_OPERATOR_HEADERS -DCPUINFO_SUPPORTED_PLATFORM=1 -DFMT_HEADER_ONLY=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DUSE_C10D_GLOO -DUSE_DISTRIBUTED -DUSE_EXTERNAL_MZCRC -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE=1 -D_OPENMP_NOFORCE_MANIFEST -Dtorch_cpu_EXPORTS -ID:\bld\pytorch-recipe_1663314310088\work\build\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\build -ID:\bld\pytorch-recipe_1663314310088\work -ID:\bld\pytorch-recipe_1663314310088\work\third_party\onnx -ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\onnx -ID:\bld\pytorch-recipe_1663314310088\work\third_party\foxi -ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\foxi -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\api -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\api\include -ID:\bld\pytorch-recipe_1663314310088\work\caffe2\aten\src\TH -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\aten\src\TH -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\build\caffe2\..\aten\src -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc -ID:\bld\pytorch-recipe_1663314310088\work\third_party\miniz-2.0.8 -ID:\bld\pytorch-recipe_1663314310088\work\third_party\kineto\libkineto\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\kineto\libkineto\src -ID:\bld\pytorch-recipe_1663314310088\work\torch\csrc\distributed -ID:\bld\pytorch-recipe_1663314310088\work\aten\..\third_party\catch\single_include -ID:\bld\pytorch-recipe_1663314310088\work\aten\src\ATen\.. -ID:\bld\pytorch-recipe_1663314310088\work\c10\.. -ID:\bld\pytorch-recipe_1663314310088\work\third_party\pthreadpool\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\cpuinfo\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fbgemm\third_party\asmjit\src -ID:\bld\pytorch-recipe_1663314310088\work\third_party\FP16\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\fmt\include -ID:\bld\pytorch-recipe_1663314310088\work\third_party\flatbuffers\include -external:ID:\bld\pytorch-recipe_1663314310088\work\build\third_party\gloo -external:ID:\bld\pytorch-recipe_1663314310088\work\cmake\..\third_party\gloo -external:ID:\bld\pytorch-recipe_1663314310088\work\third_party\protobuf\src -external:ID:\bld\pytorch-recipe_1663314310088\_h_env\Library\include -external:ID:\bld\pytorch-recipe_1663314310088\work\third_party\XNNPACK\include -external:ID:\bld\pytorch-recipe_1663314310088\work\cmake\..\third_party\eigen -external:ID:\bld\pytorch-recipe_1663314310088\work\caffe2 -external:W0 /DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -DHAVE_AVX512_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2 /DNDEBUG /w /bigobj -DNDEBUG -DCAFFE2_USE_GLOO -DTH_HAVE_THREAD /EHsc /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -O2 -DCAFFE2_BUILD_MAIN_LIB -DONNX_BUILD_MAIN_LIB -std:c++14 /showIncludes /Focaffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_mutation.cpp.obj /Fdcaffe2\CMakeFiles\torch_cpu.dir\ /FS -c D:\bld\pytorch-recipe_1663314310088\work\torch\csrc\jit\passes\remove_mutation.cpp
D:\bld\pytorch-recipe_1663314310088\work\torch\csrc\jit\passes\remove_mutation.cpp : fatal error C1085: Cannot write compiler generated file: 'D:\bld\pytorch-recipe_1663314310088\work\build\caffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_mutation.cpp.obj': No space left on device
ninja: build stopped: subcommand failed.
-- Building version 1.12.1
cmake -GNinja -DBUILD_CUSTOM_PROTOBUF=ON -DBUILD_PLATFORM=win-64 -DBUILD_PREFIX=D:\bld\pytorch-recipe_1663314310088\_build_env -DBUILD_PYTHON=True -DBUILD_TEST=False -DCMAKE_BUILD_TYPE=Release -DCMAKE_GEN=Visual Studio 16 2019 -DCMAKE_GENERATOR=Ninja -DCMAKE_INCLUDE_PATH=D:\bld\pytorch-recipe_1663314310088\_h_env\Library\include -DCMAKE_INSTALL_PREFIX=D:\bld\pytorch-recipe_1663314310088\work\torch -DCMAKE_LIBRARY_PATH=D:\bld\pytorch-recipe_1663314310088\_h_env\Library\lib -DCMAKE_PLAT=x64 -DCMAKE_PREFIX_PATH=D:\bld\pytorch-recipe_1663314310088\_h_env\Lib\site-packages;D:\bld\pytorch-recipe_1663314310088\_h_env\Library -DINSTALL_TEST=0 -DNUMPY_INCLUDE_DIR=D:\bld\pytorch-recipe_1663314310088\_h_env\lib\site-packages\numpy\core\include -DPYTHON_EXECUTABLE=D:\bld\pytorch-recipe_1663314310088\_h_env\python.exe -DPYTHON_INCLUDE_DIR=D:\bld\pytorch-recipe_1663314310088\_h_env\Include -DPYTHON_LIBRARY=D:\bld\pytorch-recipe_1663314310088\_h_env/libs/python38.lib -DTORCH_BUILD_VERSION=1.12.1 -DUSE_CUDA=0 -DUSE_MKLDNN=0 -DUSE_NEW_CMAKE_GEN_SYNTAX=1 -DUSE_NUMPY=True -DUSE_OPENMP=0 -DUSE_SYSTEM_SLEEF=ON D:\bld\pytorch-recipe_1663314310088\work
cmake --build . --target install --config Release -- -j 2
```

Did we run out of space? It compiled for 2hrs and 40min

**Comment by h-vetinari** [comment]
> `Cannot write compiler generated file: 'D:\bld\pytorch-recipe_1663314310088\work\build\caffe2\CMakeFiles\torch_cpu.dir\__\torch\csrc\jit\passes\remove_mutation.cpp.obj': No space left on device`

These things are usually intermittent, but the most realistic best case scenario is still just that the CI runs into the 6h timeout (if it doesn't blow up the memory before that)

**Comment by mikemhenry** [comment]
@hmaarrfk @h-vetinari @isuruf  What should we do next? `CFEP-03`? It looks like all the builds are just failing due to lack of disk space now

**Comment by hmaarrfk** [comment]
Sorry, this is really unfortunate, it fails at disk space.

Windows worries me in general because the automated tools are behind. @xhochy is there a large drive that we have access to on Windows + Azure?

**Comment by hmaarrfk** [comment]
Do you want to try: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/109/files#diff-878bbf6f64cef8f2c9a579cfe974788abd682c931d8dd709a11d331d148b2309R1 in order to reduce the download size?

**Comment by mikemhenry** [comment]
Trying that with https://github.com/conda-forge/pytorch-cpu-feedstock/pull/134/commits/7f9f434103bd7523c214605dbe1f600d2f3e3998
@hmaarrfk what is the best way to resolve these merge conflicts? Normally I either set a `--strategy-option` but I was wondering what you do.

**Comment by hmaarrfk** [comment]
I rebate and delete all rerender commits

**Comment by xhochy** [comment]
> Sorry, this is really unfortunate, it fails at disk space.
> 
> Windows worries me in general because the automated tools are behind. @xhochy is there a large drive that we have access to on Windows + Azure?

I think we already use the D drive which is the larger of two

**Comment by hmaarrfk** [comment]
Did you cherry-pick from my previous commit or did you rerun the script on linux? That script was pretty hacky, I didn't want to include it unless there was a gain. The gain here would be the ability to run windows.

**Comment by mikemhenry** [comment]
Sorry @hmaarrfk I just merged in your branch into this one, did you mean for me to try it a different way?

**Comment by hmaarrfk** [comment]
The PR I referred to was a script I wrote in 1.11.0 you'll have to rerun it.

https://github.com/conda-forge/pytorch-cpu-feedstock/pull/109/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR12

Im running it now.

**Comment by mikemhenry** [comment]
Just got back to this, I think I got things cleaned up but wanted to give the window builds one more try

**Comment by mikemhenry** [comment]
@hmaarrfk Any tips for building locally on windows? This is what I tried: https://gitter.im/conda-forge/conda-forge.github.io?at=6350f97786407c265b7f91fe

**Comment by mikemhenry** [comment]
I'm guessing I need to use `conda-build` but since this is for a CFEP03 I want to make sure I do it right

**Comment by hmaarrfk** [comment]
I have no tips for windows unfortunately. Could you also unskip the other architectures. We want to be ready to merge

**Comment by mikemhenry** [comment]
@hmaarrfk done! Do you know who is in charge of/wrote the scritps to build on windows?

**Comment by hmaarrfk** [comment]
What scripts to build? Didn't you wrote them?

**Comment by mikemhenry** [comment]
@hmaarrfk I mean the correct way to run these: `.ci_support/win_64_cuda_compiler_version10.2cudnn7numpy1.20python3.7.____cpython.yaml`

**Comment by ryanvolz** [comment]
> I'm guessing I need to use `conda-build` but since this is for a CFEP03 I want to make sure I do it right

(Coming here from Gitter)

I have never done a CFEP03 build, but I have built packages locally for Windows for other feedstocks. What I usually do is as simple as
```
conda build -m .ci_support\.ci_support/win_64_cuda_compiler_version10.2cudnn7numpy1.20python3.7.____cpython.yaml recipe
```
I run this in a VMWare Windows instance on my Linux machine since I'm not a native Windows user. You'll have to have `conda-build` and the appropriate Visual Studio compiler tools installed already, of course.

**Comment by mikemhenry** [comment]
Thanks @ryanvolz! That is what I was looking for!

**Comment by h-vetinari** [comment]
Broadly speaking LGTM! Thanks a lot for shepherding this PR this far! Unblocking pytorch on windows will have a great positive impact! :)

There's a bunch of clean-ups that I'd have in mind that will take more time though (allow several blas flavours, fix openmp & mkldnn, libraryize together with upstream...), plus some nits that aren't necessary to fix right away (i.e. vs2019 in cbc is obsolete), but we can fix those for 1.13.

**Comment by mikemhenry** [comment]
@hmaarrfk Looks like some of these didn't finish in time, should I try to get a smaller subset of the matrix to pass, or just build everything I can locally?.
Or is there any easy way to just re-run the failed jobs?

**Comment by hmaarrfk** [comment]
I'm not planning on rebuilding anything for linux. There may be a mismatch between build numbers, but I don't think this is an issue given that the cuda builds are 200 numbers above the non-cuda builds.

Lets just focus on the windows builds.

It is somewhat unfortunate that we are going to have to timeout on the CIs, but I think this is all we can expect for now.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,6 +25,8 @@ source:
     # We've patch out the bug in GCC that makes it incompatible
     # https://github.com/conda-forge/ctng-compilers-feedstock/blob/7ea10c2526445beb9faff23efcc6bc40fc0079b5/recipe/meta.yaml#L23=
     - patches/remove_gcc_maximum_requirements.patch 
+    # mkl-devel on Windows has suffix _dll for MKL's .lib files
+    - patches/fix_mkl_win.patch  # [win]
 
 build:
   number: {{ number }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:32]: The line below here needs to be unskipped.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by mikemhenry** [comment]
@conda-forge-admin please rerender

**Comment by mikemhenry** [comment]
Going to try and bring this back :pray: should I try and get this version going to jump to 1.13?

**Comment by h-vetinari** [comment]
> should I try and get this version going to jump to 1.13?

Whatever works, but you can jump as far ahead as you want - far more important to get anything working that what version it is exactly. Though you'll have to rebase on main, in general, but even moreso for the rerender to work.

**Comment by mikemhenry** [comment]
Looks like I have a patch to fix:
```
Applying patch: D:\a\1\s\recipe\patches/remove_gcc_maximum_requirements.patch with args:
['-Np1', '-i', 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\tmpg4en1zxv\\remove_gcc_maximum_requirements.patch.native', '--binary']
checking file torch/utils/cpp_extension.py
patch unexpectedly ends in middle of line
Hunk #1 FAILED at 49.
1 out of 1 hunk FAILED
Warning: failed to download source.  If building, will try again after downloading recipe dependencies.
Error was: 
Command '['C:\\Miniforge\\Library\\usr\\bin\\patch.exe', '--no-backup-if-mismatch', '--batch', '-Np1', '-i', 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\tmpg4en1zxv\\remove_gcc_maximum_requirements.patch.native', '--binary', '--dry-run']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "C:\Miniforge\Scripts\conda-mambabuild-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "C:\Miniforge\lib\site-packages\conda_build\api.py", line 180, in build
    return build_tree(
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 3097, in build_tree
    packages_from_this = build(metadata, stats,
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 2136, in build
    try_download(m, no_download_source=False, raise_error=True)
  File "C:\Miniforge\lib\site-packages\conda_build\render.py", line 656, in try_download
    raise RuntimeError("Failed to download or patch source. Please see build log for info.")
RuntimeError: Failed to download or patch source. Please see build log for info.
```

**Comment by hmaarrfk** [comment]
i think it is a merge conflict since all builds are failing for the same reason.

**Comment by mikemhenry** [comment]
@conda-forge-admin please rerender

**Comment by Tobias-Fischer** [comment]
Closing in favour of #231

**Pull request closed without merging**